micromasters/factories.py

```diff
@@ -2,9 +2,40 @@
 Factory for Users
 """
 from django.contrib.auth.models import User
-from factory import Sequence
+from factory import (
+    Sequence,
+    SubFactory,
+    LazyAttribute,
+)
 from factory.django import DjangoModelFactory
 from factory.fuzzy import FuzzyText
+from social_django.models import UserSocialAuth
+from backends.edxorg import EdxOrgOAuth2
+from micromasters.utils import pop_matching_keys_from_dict
+
+
+def extract_related_model_kwargs(orig_kwargs, related_model_prop_name):
+    """
+    Extracts a set of factory kwargs that refer to a related model, removes the model prefix from the keys,
+    and returns the dictionary.
+    Ex. usage:
+        user_factory_kwargs = dict(username='user_0', social_auth__provider='edx', social_auth__uid='user_0')
+        extract_related_model_kwargs(user_factory_kwargs, 'social_auth')
+        #> dict(provider='edx', uid='user_0')
+
+    Args:
+        orig_kwargs (dict): A dict of kwargs being passed to a factory generation method
+        related_model_prop_name (str): The model name as it should appear in kwargs (e.g.: 'user', 'social_auth')
+
+    Returns:
+        dict: A dict of kwargs related to a specific model (without the model prefix)
+    """
+    related_model_prefix = '{}__'.format(related_model_prop_name)
+    extracted_kwargs = pop_matching_keys_from_dict(
+        orig_kwargs,
+        lambda key: key.startswith(related_model_prefix)
+    )
+    return {k.replace(related_model_prefix, ''): v for k, v in extracted_kwargs.items()}
```

Comment:
This might do something unexpected if `related_model_prefix` exists twice in the prop name. Can you change this to `replace(related_model_prefix, '', 1)` and add a test for this edge case?